src/substrait/textplan/parser/SubstraitPlanVisitor.cpp

```diff
@@ -0,0 +1,364 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/textplan/parser/SubstraitPlanVisitor.h"
+
+#include <memory>
+
+#include "SubstraitPlanParser/SubstraitPlanParser.h"
+#include "substrait/textplan/Any.h"
+#include "substrait/textplan/Location.h"
+#include "substrait/textplan/StructuredSymbolData.h"
+
+namespace io::substrait::textplan {
+
+// Removes leading and trailing quotation marks.
+std::string extractFromString(std::string s) {
+  if (s.size() < 2) {
+    return s;
+  }
+  if (s[s.size() - 1] == '"') {
+    s.erase(s.size() - 1, 1);
+  }
+  if (s[0] == '"') {
+    s.erase(0, 1);
+  }
+  return s;
+}
+
+// NOLINTBEGIN(readability-identifier-naming)
+// NOLINTBEGIN(readability-convert-member-functions-to-static)
+
+std::any SubstraitPlanVisitor::visitPlan(
+    SubstraitPlanParser::PlanContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitPlan_detail(
+    SubstraitPlanParser::Plan_detailContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitPipelines(
+    SubstraitPlanParser::PipelinesContext* ctx) {
+  return visitChildren(ctx);
+}
+
+/*
+ * Creates symbols in order of discovery, since the first symbol encountered is
+ * the bottom of the chain, create children nodes first, then create the symbol.
+ * Visits children to accomplish that.
+ * Children know their parents allowing them to document the relationship.
+ */
+std::any SubstraitPlanVisitor::visitPipeline(
+    SubstraitPlanParser::PipelineContext* ctx) {
+  if (ctx->pipeline() != nullptr) {
+    visitPipeline(ctx->pipeline());
+  }
+
+  auto parentCtx = (antlr4::ParserRuleContext*)ctx->parent;
+  if (dynamic_cast<SubstraitPlanParser::PipelinesContext*>(parentCtx)) {
+    parentCtx = nullptr;
+  }
+
+  auto rel = ANY_CAST(std::string, visitRelation_ref(ctx->relation_ref()));
+  symbolTable_->defineSymbol(
+      rel, Location(ctx), SymbolType::kRelation, std::nullopt, parentCtx);
+
+  return rel;
+}
+
+std::any SubstraitPlanVisitor::visitExtensionspace(
+    SubstraitPlanParser::ExtensionspaceContext* ctx) {
+  const std::string& uri = ctx->URI()->getText();
+  // TODO -- Transition to querying the symbol table.
```

Comment:
Shall we create an issue and reference it here?

Reply:
Added #42.